### PR TITLE
Add permalink to timestamp

### PR DESCRIPTION
**Pull request by andreistanciu24**
Please make sure you've read the [pull request](http://docs.mattermost.com/developer/contribution-guide.html#preparing-a-pull-request) section of our [code contribution guidelines](http://docs.mattermost.com/developer/contribution-guide.html).

When filling in a section please remove the help text and the above text.

#### Summary
Timestamps on messages should be permalinks

#### Ticket Link
https://github.com/mattermost/platform/issues/5183

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] Added API documentation (required for all new APIs)
- [ ] All new/modified APIs include changes to the drivers
- [ ] Has enterprise changes (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

**Comment by grundleborg** [comment]
Hi @andreistanciu24 

Thanks very much for contributing this pull request. Would you be able to rebase it on top of the current master branch and do a forced push? The continuous integration server is having issues building it due to a recent commit on master, and rebasing this PR should fix the issue.

**Comment by andreistanciu24** [comment]
Hi @grundleborg 

I did a rebase on top of the current master branch but I'm not authorized to push it to your master branch. I only pushed it to my forked repository master branch. Is there anything else I can do?

**Comment by grundleborg** [comment]
@andreistanciu24 you don't need to push master, just the branch for this PR. So assuming your local `master` branch is now up to date with the latest `master` from this repository, something like:

    $ git checkout PLT-5376
    $ git rebase master
    $ git push origin -f PLT-5376

should do the trick.

**Comment by andreistanciu24** [comment]
Done

**Comment by esethna** [comment]
@andreistanciu24 thanks for this PR! Looking great, just a few notes:

1. Can we maintain the grey color on the permalinks and then just add the underline on hover?

2. If you are in mobile web view, then expand your window to be larger, the links do not appear until a page refresh. Similarly, when you scale your window down from large size to mobile view, the links do not disappear until a page refresh or a long delay.

3. On the desktop app (tested on Mac), when you click the timestamp, the permalink opens in a new Mattermost desktop tab - can we instead get it to open in the users primary browser?

Other than that it's looking awesome. Thanks for your contribution!

**Comment by andreistanciu24** [comment]
@esethna the first 2 points are done. I pushed them on my branch. Regarding the 3rd one, can you please give me a link to the right section in the documentation? I don't know how to install/setup/test the desktop app, and I didn't find anything in the docs. If you think it's better maybe you can merge the branch as it is and file another issue with just the desktop app.

File: webapp/components/post_view/components/post_time.jsx [hunk 1]
```diff
@@ -6,26 +6,40 @@ import React from 'react';
...
+import {Link} from 'react-router/es6';
+import TeamStore from 'stores/team_store.jsx';
 
 export default class PostTime extends React.Component {
     constructor(props) {
         super(props);
 
         this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
+        this.state = {
+            currentTeamDisplayName: TeamStore.getCurrent().display_name,
```
Comment (coreyhulen) [inline webapp/components/post_view/components/post_time.jsx]: I think this only works if `display_name` matches `name`.  I think you want `name` here.

Comment (andreistanciu24) [inline webapp/components/post_view/components/post_time.jsx]: Done. Thanks !

**Comment by esethna** [comment]
Thanks @andreistanciu24, the desktop app install docs are [here](https://docs.mattermost.com/install/desktop.html), would you mind attempting to install and repo? The current behaviour for website links in the desktop app is that they open as a new tab in the users primary browser, so we would want to match that with the permalinks,

**Comment by andreistanciu24** [comment]
I think that those docs are for the production version of the app. I can't find any instructions about the development environment tho. I'm not sure how my sources for the webapp are related to the sources for a desktop app..

**Comment by esethna** [comment]
@andreistanciu24, the Mattermost desktop apps are based on Electron, which integrates the Chrome engine within the standalone application. See the [deskop repo here](https://github.com/mattermost/desktop). 

@coreyhulen would you know if the change to get permalinks in the desktop app to open in a browser tab, rather than a new desktop app window, is a fix that needs to be made on this repo or the desktop repo?

**Comment by coreyhulen** [comment]
@esethna not sure, do normal permalinks work correctly or open in a new window on the desktop app?  If that's the case then it's probably unrelated to this change.  If old/normal links work as expected then it's probably something in this PR.

**Comment by esethna** [comment]
Currently opening a permalink in a message in the desktop app changes the active window to archive view, but external website links open in a new browser tab. 

This PR behaves differently, where clicking on the timestamps opens the permalink in a different desktop app window, rather than a new browser tab, which would be expected. So I'm not sure if this change needs to be addressed on this repo or the desktop repo. Maybe @jasonblais would know someone in the desktop community that can help?

**Comment by coreyhulen** [comment]
If that's the case then its probably `target='_blank'` in this PR ?  We should copy how the other permalinks render their tags.

**Comment by andreistanciu24** [comment]
@coreyhulen target='_blank' is already on the link component as requested:

Clicking on the timestamp should:

Web: Open the permalink of the message in a new tab

You're saying we should remove it? 
Anyway I don't see the link between the webapp and the desktop app. From my understanding they are different apps on different repositories. How does a change in this repo affects the app on the other repo? Forgive me if it's a dumb question.

**Comment by coreyhulen** [comment]
Yes, I was suggesting removing the `target='_blank'` and make it function like other permalinks.  @esethna are we changing the functionality where it's suppose to open in a new browser tab for time permalinks?  Not sure that makes sense.  It should just follow the permlink like the others.

desktop app is an electron app.  It just loads and runs the webapp.  So a change in the webapp is also a change in the desktop app.

**Comment by andreistanciu24** [comment]
sure thing. I'll fix them right away

**Comment by andreistanciu24** [comment]
Done. I pushed it on my branch

**Comment by jasonblais** [comment]
Hey  @andreistanciu24, looks like there's a build failure - can you help take a look?

Thanks for your contribution! Great work 👍

**Comment by andreistanciu24** [comment]
Done. It was an error with a duplicated import. I also updated my branch with the latest updates from the upstream

**Comment by esethna** [comment]
Hi @andreistanciu24, thanks for the tweaks! Sorry, my earlier comment may not have been clear,

1. For the timestamp to the left of consecutive posts, can we please only underline when hovering over the timestamp itself?
image

Other than that this PR is looking really great! Very excited about this feature

**Comment by andreistanciu24** [comment]
Done. You were perfectly clear, I just misread it :) sorry for that

**Comment by andreistanciu24** [comment]
Thanks a lot !

**Comment by grundleborg** [comment]
@coreyhulen I think this is ready for you to review again.

**Pull request merged**